class.go

```diff
@@ -41,21 +44,29 @@ func (q HtbClassAttrs) String() string {
 // Htb class
 type HtbClass struct {
 	ClassAttrs
-	Rate    uint64
-	Ceil    uint64
+	Rate    nl.TcRateSpec
+	Ceil    nl.TcRateSpec
 	Buffer  uint32
 	Cbuffer uint32
 	Quantum uint32
 	Level   uint32
 	Prio    uint32
+	Rtab    [256]uint32
+	Ctab    [256]uint32
 }
 
-func NewHtbClass(attrs ClassAttrs, cattrs HtbClassAttrs) *HtbClass {
+func NewHtbClass(attrs ClassAttrs, cattrs HtbClassAttrs) (*HtbClass, error) {
```

Comment:
So the error being returned is mainly due to `CalcRtable`. I had thought about it, but from a usage standpoint, I think it is better to provide the rates as attribute rather than the opaque `nl.TcRateSpec`.

That being said, I would be rather kin on aligning this with the rest of the code base for the sake of uniformity though. Will check how this is handle in other places.